controllers/buildpipeline/buildpipeline_controller.go

```diff
@@ -88,11 +88,15 @@ func (r *Reconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Resu
 	})
 	if err != nil {
 		if errors.IsNotFound(err) {
+			logger.Info("Failed to get component for build pipeline - skipping processing, it's not an appstudio pipeline", "name", pipelineRun.Name, "namespace", pipelineRun.Namespace)
```

Comment:
Thinking about this, shouldn't the non-appstudio pipelineRuns be missing the component label in the first place?
In that case, the `IsNotFound` error shouldn't be possible for those pipelineRuns, the component should be set to nil [at this point](https://github.com/redhat-appstudio/integration-service/blob/eb807b310184c9b9a29a910140edfd473ef8fc9b/controllers/buildpipeline/buildpipeline_controller.go#L86) instead.

Reply:
That's true, we also handle that case [here](https://github.com/redhat-appstudio/integration-service/blob/eb807b310184c9b9a29a910140edfd473ef8fc9b/controllers/buildpipeline/buildpipeline_controller.go#L99) but if we want to we can do some error handling above instead of just failing silently

Reply:
I'm confused here, it seems that invalid component case is already handled well, and we cannot really say if that is caused by not being RHTAP pipeline or removal of component.

Then if component label is missing, we just continue processing it, there is just single condition that creation of snapshot is skipped. Why do we even process PLR without component and don't just stop it here in reconcile initialization?

Reply:
Agreed, I think we should just stop processing if the component label for the build pipelineRun is missing.

Reply:
Based on this conversation and the discussion on Slack from earlier today I've made the following changes
1. Stop reconciliation if `GetComponentFromPipelineRun()` returns `nil, nil`
2. Remove the check to verify that the component is non-nil before calling `GetApplicationFromComponent()`.  It's no longer needed
3. Replace the call to `helpers.HandleLoaderError()` with a logger call and a return. That function just checks for `errors.IsNotFound()` and logs/returns accordingly, which we do in the lines above already.
4. Remove the couple lines of dead code in the buildpipeline adapter that checks if the component is nil

This turned from a bugfix into more of a cleanup PR and I'll update the commit message accordingly

Reply:
Thanks, I think this change will correctly handle the original issue in [STONEINTG-645](https://issues.redhat.com/browse/STONEINTG-645).